src/character.cpp

```diff
@@ -1009,6 +1009,32 @@ std::vector<item_location> Character::find_ammo( const item &obj, bool empty, in
     return res;
 }
 
+std::vector<item_location> Character::find_reloadables()
+{
+    std::vector<item_location> reloadables;
+
+    visit_items( [this, &reloadables]( item * node ) {
+        if( node->is_holster() ) {
+            return VisitResponse::NEXT;
+        }
+        if( node->is_gun() && !node->magazine_compatible().empty() ) {
+            if( node->magazine_current() == nullptr ||
+                node->ammo_remaining() < node->ammo_capacity() ) {
+                reloadables.push_back( item_location( const_cast<Character &>( *this ), node ) );
+                return VisitResponse::SKIP;
+            }
+        }
+        if( ( node->is_magazine() || node->is_bandolier() ||
+              ( node->is_gun() && node->magazine_integral() ) ) &&
+            node->ammo_remaining() < node->ammo_capacity() ) {
```

Comment:
Is it feasible to rewrite the conditions into one complex and apply formatting to get something like this:

```c++
const bool is_reloadable = // The complex condition follows here.

if( is_reloadable ) {
    reloadables.emplace_back( const_cast<Character &>( *this ), node );
}

return VisitResponse::SKIP;
```

Reply:
It ends up entirely too long to go into one statement, but it does clean it up a good bit to just have a boolean and accumulate it.